doc: add security recommendation about *notify config params (#23850)
bitcoin/bitcoin: doc/bitcoin-conf.md

macOS | `$HOME/Library/Application Support/Bitcoin/` | `/Users/username/Library/Application Support/Bitcoin/bitcoin.conf`

You can find an example bitcoin.conf file in [share/examples/bitcoin.conf](../share/examples/bitcoin.conf).

### Command-line options and Security

> `-startupnotify`, `-shutdownnotify`, `-blocknotify`, `-walletnotify` and `-alertnotify` configuration parameters accept shell commands to be executed based on different events. These are useful for notifying the user of certain events; however, they can be misused by malicious software or social engineering attacks. Restrict scripts and commands in the shell/system used for bitcoind to avoid running potentially unsafe scripts.

Reviewer: 1. You shall add a line break after each sentence. This makes the sentences easier to distinguish and read in the source file while keeping the rendered markdown unchanged.

```suggestion
`-startupnotify`, `-shutdownnotify`, `-blocknotify`, `-walletnotify`, and `-alertnotify` configuration parameters accept shell commands to be executed based on different events. 
These are useful for notifying the user of certain events; however, they can be misused by malicious software or social engineering attacks. 
Restrict scripts and commands in the shell/system used for bitcoind to avoid running potentially unsafe scripts.
```

2. The last sentence in the paragraph is hard to understand. The sentence suggests restricting the use of scripts and commands irrespective of your expertise in using them. I suggest an alternative for the last sentence to make it more straightforward.

```suggestion
`-startupnotify`, `-shutdownnotify`, `-blocknotify`, `-walletnotify`, and `-alertnotify` configuration parameters accept shell commands to be executed based on different events. These are useful for notifying the user of certain events; however, they can be misused by malicious software or social engineering attacks. Please do not run any script and command before sufficiently knowing what they do to avoid the potential risk of harm.
```

Author: The difference between **suggesting to restrict scripts** and **asking the user to examine each script** before running:

Alice might be able to identify if something is malicious however this attack or security recommendation was never meant for Alice. Bob cannot identify if something is malicious and runs few commands someone suggested to use with *notify config param to fix one of the issues. Bob thinks he is _doing the right thing_ but he is running a command which could end up in losing all his money.

**How to avoid this?** _Don't examine each script._ Create some restrictions that blocks any suspicious commands, scripts, connections etc. Unfortunately we don't have any such feature in Bitcoin Core so users have to do it at system level.

Examples:

1. PowerShell has execution polices which can be used to restrict scripts: https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_execution_policies?view=powershell-7.2

2. Macros can be used to run malicious scripts in Word documents. However we have settings to disable it (restriction). Attackers still try weird things using social engineering and many are successful.

![image](https://user-images.githubusercontent.com/13405205/147384691-b5a4d76d-010a-4ce2-beb0-76d859a4941f.png)

https://www.ncsc.gov.uk/guidance/macro-security-for-microsoft-office

> `-startupnotify`, `-shutdownnotify`, `-blocknotify`, `-walletnotify` and `-alertnotify` configuration parameters accept shell commands to be executed based on different events. These are useful for notifying the user of certain events; however, they can be misused by malicious software or social engineering attacks. Restrict scripts and commands in the shell/system used for bitcoind to avoid running potentially unsafe scripts.

Reviewer 2: It might be better to use`-*notify` instead. This removes the need to constantly update this doc when a new `-*notify` parameter is added. (Just a suggestion, feel free to ignore)

```suggestion
`-*notify` configuration parameters accept shell commands to be executed based on different events. These are useful for notifying the user of certain events; however, they can be misused by malicious software or social engineering attacks. Restrict scripts and commands in the shell/system used for bitcoind to avoid running potentially unsafe scripts.
```

> `-startupnotify`, `-shutdownnotify`, `-blocknotify`, `-walletnotify` and `-alertnotify` configuration parameters accept shell commands to be executed based on different events. This is useful for notifying the user of certain events, however it can be misused by malicious software or attackers using social engineering. Restrict scripts and commands in the shell/system used for bitcoind to avoid running potentially unsafe scripts.

Reviewer: AFAIK #23395 is not yet merged, and hence `-shutdownnotify` has not yet been added to master. So you should remove that from the docs.

```suggestion
`-startupnotify`, `-blocknotify`, `-walletnotify`, and `-alertnotify` configuration parameters accept shell commands to be executed based on different events. These are useful for notifying the user of certain events; however, they can be misused by malicious software or social engineering attacks. Restrict scripts and commands in the shell/system used for bitcoind to avoid running potentially unsafe scripts.
```

Author: > AFAIK #23395 is not yet merged, and hence -shutdownnotify has not yet been added to master. So you should remove that from the docs.

This PR can be considered a part of #23395 so they should be merged together. It adds docs and I have suggested to add the link for this doc in release notes. There is no code.

Fix other errors in grammar in https://github.com/bitcoin/bitcoin/commit/78a364cf1a3ef89b8592c1fd81e794b535cfcd17
